Common/Helpers/BlobStorageClient.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers
+{
+    /// <summary>
+    /// Helper methods, related to blob storage.
+    /// </summary>
+    public class BlobStorageClient : IBlobStorageClient
+    {
+        private CloudStorageAccount _storageAccount;
+        private CloudBlobContainer _container;
+        private CloudBlockBlob _blob;
+        private readonly CloudBlobClient _blobClient;
+        private readonly string _containerName;
+        private readonly string _blobName;
+
+        public BlobStorageClient(string connectionString, string containerName, string blobName)
+        {
+            _storageAccount = CloudStorageAccount.Parse(connectionString);
+            _blobClient = _storageAccount.CreateCloudBlobClient();
+            _containerName = containerName;
+            _blobName = blobName;
+        }
+        
+        public async Task UploadFromByteArrayAsync(byte[] buffer, int index, int count, AccessCondition accessCondition, BlobRequestOptions options, OperationContext operationContext)
+        {
+            await this.CreateCloudBlockBlobAsync();
+            await _blob.UploadFromByteArrayAsync(
```

Comment:
CreateCloudBlockBlobAsync should not update state of the class. Instead return the blob and UploadFromByteArrayAsync should use the blob instance directly.